src/pages/AssignServerRoles.js

```diff
@@ -978,6 +999,63 @@ class AssignServerRoles extends BaseWizardPage {
     }
   }
 
+  /**
+   * When a server is edited (which is only possible in from the assigned servers page
+   * on the right), update the details of the matching entry in the list that backs
+   * the available servers (on the left).  This ensures that the correct information
+   * is persisted to the discovered server store, and also ensures that no information
+   * is dropped when unassigning and reassigning servers to roles.
+   */
+  updateServerForEditServer = (server) => {
+    for (let list of ['rawDiscoveredServers', 'serversAddedManually']) {
+      let idx = this.state[list].findIndex(s => server.id === s.id);
+      if (idx >= 0) {
+        let updated_server;
+        this.setState(prev => {
```

Comment:
wow, the syntax of this whole statement takes a minute to understand

Reply:
I just restored the functions that were refactored a couple of times...:).